### PR TITLE
Feat/user profile first iteration

### PR DESCRIPTION
**Pull request by mat10tng**

### CODE REVIEW

File: src/app/component/Button/Button.js [hunk 1]
```diff
@@ -1,44 +1,17 @@
 import React from "react";
 import PropTypes from "prop-types";
-import StyledButton, { StyledIcon } from "./Button.style";
+import { Button } from "@material-ui/core";
+import { makeStyles, useTheme } from "@material-ui/core/styles";
+const useStyles = makeStyles((theme) => ({}));
+
+export function CustomizeButton({ children, text, ...rest }) {
+  const classes = useStyles();
+  const theme = useTheme();
```
Comment (azmainamin) [inline src/app/component/Button/Button.js:9]: Unused. Remove

File: src/app/component/Button/ButtonGroup.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+import { ButtonGroup } from "@material-ui/core";
+import { makeStyles, useTheme } from "@material-ui/core/styles";
+
+const useStyles = makeStyles((theme) => ({}));
+export function CustomizeButton({ children, ...rest }) {
+  const classes = useStyles();
+  const theme = useTheme();
+
+  return (
+    <ButtonGroup className={classes.root} {...rest}>
```
Comment (azmainamin) [inline src/app/component/Button/ButtonGroup.js:12]: classes don't have a root object. Any particular reason for no root styling?

File: src/app/component/SMSLogin/SMSLogin.jsx [hunk 3]
```diff
@@ -50,7 +50,7 @@ function SMSLogin({ loginWithFacebook, loginWithGoogle, handlePhoneNumberChange,
                 id="phone-number"
                 label="Phone Number"
                 type="text"
-                helperText="Format: +1 2345678910"
+                helperText="Format: +1 540-319-7017, verificationCode: 123456"
```
Comment (azmainamin) [inline src/app/component/SMSLogin/SMSLogin.jsx:53]: Umm... I would like it if my phone number is not on the helper text. lol

File: src/app/page/UserProfile/LinkGoogleAccount.jsx [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    width: "48px",
+  },
+}));
+
+const LinkGoogleAccount = ({ data, auth, errorHandler }) => {
+  const classes = useStyles();
+  async function onGoogleClick() {
+    var provider = new auth.GoogleAuthProvider();
+    try {
+      const credential = await auth.currentUser.linkWithPopup(provider).then(/* redirect */);
```
Comment (azmainamin) [inline src/app/page/UserProfile/LinkGoogleAccount.jsx:25]: Should we redirect to some place? its just a comment for now

File: src/app/page/UserProfile/LinkPhoneAccount.jsx [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+      </Button>
+    </>
+  ) : (
+    <Grid container direction="column" spacing={2}>
+      <Grid item>
+        <TextField
+          id="phone-number"
+          label="Phone Number"
+          type="text"
+          helperText="Format: +1 540-319-7017, verificationCode: 123456"
```
Comment (azmainamin) [inline src/app/page/UserProfile/LinkPhoneAccount.jsx:54]: Same comment. Please remove my number :P

File: src/app/page/UserProfile/LinkPhoneAccount.jsx [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+      >
+        <EditIcon /> Change Phone Number
+      </Button>
+
+      <Button id="phone-number-link" onClick={onPhoneClick} aria-label="change phone number">
+        Disconnect
+      </Button>
+    </>
+  ) : (
+    <Grid container direction="column" spacing={2}>
```
Comment (azmainamin) [inline src/app/page/UserProfile/LinkPhoneAccount.jsx:48]: We can reuse SMSLogin.jsx here.

File: src/app/page/UserProfile/UserProfile.js [hunk 7]
```diff
@@ -1,32 +1,182 @@
 import React from "react";
+import { useSelector } from "react-redux";
 import { withRouter } from "react-router-dom";
 import PropTypes from "prop-types";
 import { Page, Card } from "../../layout";
-import { Chip } from "../../component";
-import { StyledImage, StyledHr } from "./UserProfile.style";
-import { Typography } from "@material-ui/core";
+import { Chip, Button, H5 } from "../../component";
```
Comment (azmainamin) [inline src/app/page/UserProfile/UserProfile.js:6]: We definitely need to refactor this file into multiple files later.

File: src/app/page/UserProfile/LinkPhoneAccount.jsx (hunk 6 above)
Comment (mat10tng) [reply src/app/page/UserProfile/LinkPhoneAccount.jsx:48]: Yeah, not possible, there is going to be unlink and just other stuff as well

File: src/app/page/UserProfile/UserProfile.js (hunk 7 above)
Comment (mat10tng) [reply src/app/page/UserProfile/UserProfile.js:6]: yeah, I am still working on that, it is a quite a bit left

**Pull request closed without merging**